core/templates/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -17,247 +17,330 @@
-          initExplorationPreviewPlayer(callback);
-        } else {
-          initExplorationPlayer(callback);
-        }
-      },
-      initializeQuestionPlayer: function(
-          config, successCallback, errorCallback) {
-        PlayerTranscriptService.init();
-        initQuestionPlayer(config, successCallback, errorCallback);
-      },
-      getCurrentEngineService: function() {
-        return currentEngineService;
-      },
-      isInPretestMode: function() {
-        return explorationMode === EXPLORATION_MODE.PRETEST;
-      },
-      isInQuestionMode: function() {
-        return explorationMode === EXPLORATION_MODE.PRETEST ||
-        explorationMode === EXPLORATION_MODE.QUESTION_PLAYER;
-      },
-      isInQuestionPlayerMode: function() {
-        return explorationMode === EXPLORATION_MODE.QUESTION_PLAYER;
-      },
-      isInStoryChapterMode: function() {
-        return explorationMode === EXPLORATION_MODE.STORY_CHAPTER;
-      },
-      getPretestQuestionCount: function() {
-        return QuestionPlayerEngineService.getPretestQuestionCount();
-      },
-      moveToExploration: function(callback) {
-        if (
-          UrlService.getUrlParams().hasOwnProperty('story_url_fragment') &&
-          UrlService.getUrlParams().hasOwnProperty('node_id')) {
-          setStoryChapterMode();
-        } else {
-          setExplorationMode();
-        }
-        ExplorationEngineService.moveToExploration(callback);
-      },
-      getLanguageCode: function() {
-        return currentEngineService.getLanguageCode();
-      },
-      recordNewCardAdded: function() {
-        return currentEngineService.recordNewCardAdded();
-      },
-      get onTotalQuestionsReceived() {
-        return _totalQuestionsReceivedEventEmitter;
-      },
-      get onPlayerStateChange() {
-        return _playerStateChangeEventEmitter;
-      },
-      get onOppiaFeedbackAvailable() {
-        return _oppiaFeedbackAvailableEventEmitter;
+    this.storyUrlFragment = this.urlService.getStoryUrlFragmentFromLearnerUrl();
+  }
+
+  initializeExplorationServices(
+      returnDict: FetchExplorationBackendResponse,
+      arePretestsAvailable: boolean,
+      callback: (stateCard: StateCard, str: string) => void
+  ): void {
+    // For some cases, version is set only after
+    // ReadOnlyExplorationBackendApiService.loadExploration() has completed.
+    // Use returnDict.version for non-null version value.
+    this.statsReportingService.initSession(
+      this.explorationId, returnDict.exploration.title, returnDict.version,
+      returnDict.session_id,
+      this.urlService.getCollectionIdFromExplorationUrl());
+    this.playthroughService.initSession(
+      this.explorationId, returnDict.version,
+      returnDict.record_playthrough_probability);
+    this.playerCorrectnessFeedbackEnabledService.init(
+      returnDict.correctness_feedback_enabled);
+    this.explorationEngineService.init(
+      returnDict.exploration, returnDict.version,
```

Comment:
@DubeySandeep returnDict.exploration is of the type `ReadOnlyExplorationBackendDict` and init function expects type ExplorationBackendDict.